worlds/pokemon_rb/__init__.py

```diff
@@ -0,0 +1,251 @@
+        item_pool = []
+        for location in locations:
+            if "Hidden" in location.name and not self.world.randomize_hidden_items[self.player].value:
+                continue
+            if "Rock Tunnel B1F" in location.region and not self.world.extra_key_items[self.player].value:
+                continue
+            if location.name == "Celadon City - Mansion Lady" and not self.world.tea[self.player].value:
+                continue
+            item = self.create_item(location.original_item)
+            if location.event:
+                self.world.get_location(location.name, self.player).place_locked_item(item)
+            elif ("Badge" not in item.name or self.world.badgesanity[self.player].value) and \
+                    (item.name != "Oak's Parcel" or self.world.old_man[self.player].value != 1):
+                item_pool.append(item)
+        self.world.random.shuffle(item_pool)
+
+        self.world.itempool += item_pool
+
+        process_pokemon_data(self)  # strange place for this, but it is the only place that works for now
+
+    def pre_fill(self):
+
+        process_wild_pokemon(self)
+        process_static_pokemon(self)
+
+        if self.world.old_man[self.player].value == 1:
+            item = self.create_item("Oak's Parcel")
+            locations = []
+            for location in self.world.get_locations():
+                if location.player == self.player and location.item is None and location.can_reach(self.world.state) \
+                        and location.item_rule(item):
+                    locations.append(location)
+            self.world.random.shuffle(locations)
+            location = locations.pop()
+            location.place_locked_item(item)
+
+
+
+        if not self.world.badgesanity[self.player].value:
+            self.world.non_local_items[self.player].value -= self.item_name_groups["Badges"]
+            for i in range(5):
+                try:
+                    badges = []
+                    badgelocs = []
+                    for badge in ["Boulder Badge", "Cascade Badge", "Thunder Badge", "Rainbow Badge", "Soul Badge",
+                                  "Marsh Badge", "Volcano Badge", "Earth Badge"]:
+                        badges.append(self.create_item(badge))
+                    for loc in ["Pewter Gym - Brock 1", "Cerulean Gym - Misty 1", "Vermilion Gym - Lt. Surge 1",
+                                "Celadon Gym - Erika 1", "Fuchsia Gym - Koga 1", "Saffron Gym - Sabrina 1",
+                                "Cinnabar Gym - Blaine 1", "Viridian Gym - Giovanni 1"]:
+                        badgelocs.append(self.world.get_location(loc, self.player))
+                    state = self.world.get_all_state(False)
+                    self.world.random.shuffle(badges)
+                    self.world.random.shuffle(badgelocs)
+                    fill_restrictive(self.world, state, badgelocs.copy(), badges.copy(), True, True)
```

Comment:
```suggestion
                    fill_restrictive(self.world, state, badgelocs, badges, True, True)
```
shouldn't need the copy, as you already create a fresh copy for each iteration

Reply:
Badges doesn't need to be copied, but I have badgelocs copied so I have the original for the exception handling to remove the placed badges